test/junit/scala/concurrent/impl/DefaultPromiseTest.scala

```diff
@@ -341,4 +342,28 @@ class DefaultPromiseTest {
     }
   }
 
+  /** Test correct promise completion on RejectedExecutionException. */
+  @Test
+  def testRejectedExecutionException = {
+    class CustomForkJoinPool(parallelism: Int) extends ForkJoinPool(parallelism) {
```

Comment:
The FJP part isn't needed, so perhaps something like this:

~~~scala
val ec = ExecutionContext.fromExecutor(new Executor { override def execute(r: Runnable): Unit = throw new RejectedExecutionException("Queue capacity exceeded") })
~~~